blmath/util/json.py

```diff
@@ -0,0 +1,46 @@
+'''
+If an object defines for_json(), it will be serialized as what that returns.
+    class SerializableThing(object):
+        def for_json(self):
+            return "|SerializableThing|"
+
+'''
+
+from baiji.serialization import json
+from baiji.serialization.json import JSONDecoder
+
+class BlmathJSONDecoder(JSONDecoder):
+    SUPPORTED_TYPES = {'value': 'blmath.value.Value'}
+    def __init__(self):
+        super(BlmathJSONDecoder, self).__init__()
+        self.register(self.decode_supported_types)
+
+    def decode_supported_types(self, dct):
```

Comment:
done -- also removed a test for an error that the simplified decoder doesn't worry about.